Feature: user login

Kirill9m opened the pull request:
### Description
I’ve added the authentication system and user verification.
As a small bonus:
A few admin pages + screenings and movies submitters
All the models in the database have been added temporarily, so feel free to change them as you see fit!

Unfortunately, the token is currently only stored on the frontend, so protecting pages and components can only be done on the frontend. This isn’t very critical though.

**The entire design and styling were developed by @jennymakki .
Thanks for this part of the work!**


Fixes # (issue)

### Type of change
- [ ] Bug fix
- [x] New feature
- [x] Breaking change
- [ ] Documentation update

### How Has This Been Tested?

- [x] All tests have been done manually and visually so far
 **Will definitely be added later.**

### Checklist:
- [x] My code follows the project's code style.
- [] I have performed a self-review of my own code.
- [x] I have updated the documentation if necessary.
- [ ] New and existing tests pass with my changes.

### Additional Notes
### Instructions for Using the Auth Context

Import the useAuth Hook
First, import the useAuth hook in your component:

    import { useAuth } from "src/components/user/AuthData";

Destructure Auth Data
Then, use the useAuth hook to get the user data, authentication status, loading state, and other necessary values:

    const { userData, isLoggedIn, isLoading, logout, login, isAdmin, token } = useAuth();

Now, you can use these values throughout your component as needed!

### Example: Protecting Pages for Logged-in Users

If you want to protect specific pages so that only logged-in users can access them, you can implement a simple check using the useAuth hook and useEffect:

    import { useAuth } from "src/components/user/AuthData";
    import { useRouter } from "next/navigation";
    import { useEffect } from "react";

    const { userData, isLoggedIn, isLoading, logout, isAdmin } = useAuth();
    const router = useRouter();

    useEffect(() => {
      if (!isLoading && !isLoggedIn) {
       router.push("/"); // Redirect to the home page if the user is not logged in
      }
    }, [isLoggedIn, isLoading, router]);

    if (isLoading) return <p>Loading page data...</p>;
    if (!isLoggedIn) return <p>Access Denied. You are not authorized to view this page.</p>;

### Example: Backend Token Validation

On the backend, you can check the user's token to validate their authentication. Here's how you might check the token:

    import { useAuth } from "src/components/user/AuthData";
    import { NextResponse } from "next/server"; // Import NextResponse

    const authenticatedUser = await checkAuth(req);
       if (!authenticatedUser) {
         return NextResponse.json({ message: "Unauthorized" }, { status: 401 });
       }

      // Optionally, check if the user is an admin
     const isAdmin = authenticatedUser.role === "admin";
       if (!isAdmin) {
        return NextResponse.json(
         { message: "You don't have the right to use this feature!" },
         { status: 403 }
        );
      }

### Example: Hiding Components Based on Login Status

You can also conditionally render components based on whether the user is logged in. Here's an example of how to hide or show content depending on the login status:

    const { isLoggedIn } = useAuth();
    {!isLoggedIn ? (
      <p>Show this data to non-logged-in users</p>
       ) : (
      <p>Hide this data for logged-in users</p>
     )}

jennymakki reviewed (approved):
I've checked out your branch and it looks awesome! Everything regarding log in works smoothly and the error-handling is great. The code is nicely written and I will absolutely take inspiration from your way of writing commits - it makes it much easier to remember what and when you did something. Impressive work! 

I've got some feedback about UX related things: 
- The "Log-out icon" has fully replaced the "Logga in" button, even though you're not logged in yet.
- Some hover effects, like a pointer, to the "log out" icon and the "Inte medlem? Registrera dig här!" would enhance the experience and make it clearer that they are clickable. 
- Maybe after a person has logged in they should right away be directed to their membership-page? (Just a thought, don't know what would be best)
- When you are on the membership page I think it might be clearer if the "edit" button and the "log out" button would change places? I'll attach an image for clarity. 
![Skärmbild 2025-05-13 102707](https://github.com/user-attachments/assets/35e4cb0f-6442-4958-a6d6-6b72100228cf)

Thanks for the praise about the design! I'd be happy to look into these UX things I mentioned as feedback above. 😄

Klumpish reviewed (approved):
great guide! was super easy to understand, it was fun talking with you about the code, and using it in my own PR. everything works.

Kirill9m commented:
@jennymakki 
![image](https://github.com/user-attachments/assets/024d32bd-fe07-4f6f-8029-e6246f7c2ee8)
![image](https://github.com/user-attachments/assets/264670f6-99c6-42a0-a246-5c562dc6200c)
Brought the login button back 😉
Added hover effect to the text and changed the profile icon.

PS.
All discussions were held in Discord and the design was updated collaboratively.
